### ISSUE TITLE
Recompiling a RegExp at runtime using \p fails

### ISSUE THREAD

**Issue by atoomic** [CpanelInc/perl-compiler#65]
When recompiling a regexp using \p the recompiled RegExp is incorrect

```perl
our ( $q, $myre );
BEGIN { $q = qr[\p{IsWord}] }
eval q/$myre = qr[^$q]/; # add ^ to force the RegExp to be recompiled
print qq[match ok\n] if q[hello] =~ $myre;
```

The compiled version output nothing (no segfault) whereas we should expect `match ok` output

tested with HEAD @ 5f74704e4e03d7130d2fafb4cce9f3575e4b8bf9

**Comment by atoomic** [comment]
reported upstream via https://github.com/rurban/perl-compiler/issues/411

**Comment by toddr** [comment]
This was fixed in 4993d9be79f1b931a73fa8d434d2f0f7597e64ae
